pkg/watermark/generic/generic.go

```diff
@@ -7,14 +7,14 @@ import (
 	"github.com/numaproj/numaflow/pkg/watermark/store"
 )
 
-// NewGenericEdgeFetch returns a Fetcher, where bufferName is the from buffer of the vertex that is currently processing.
-// fetcher is an interface for retrieving both the heartbeat and the offset watermark timeline (Vn-1 vertex).
+// NewGenericEdgeFetch returns a Fetcher, which fetches watermarks from bufferName by consulting the corresponding storeWatcher
 func NewGenericEdgeFetch(ctx context.Context, bufferName string, storeWatcher store.WatermarkStoreWatcher) fetch.Fetcher {
 	processorManager := fetch.NewProcessorManager(ctx, storeWatcher)
 	return fetch.NewEdgeFetcher(ctx, bufferName, processorManager)
 }
 
 // NewGenericSourceFetch returns Fetcher, where sourceBufferName is the source buffer of the source vertex.
+// TODO remove this - we no longer need a source fetcher since we now look at the out edge to determine watermark for a vertex.
```

Comment:
Isn't it being used in source vertex?

Reply:
You are right. I will remove this comment.